### ISSUE TITLE
kfctl kustomize, ksonnet - package manager info should be integrated into config files

### ISSUE THREAD

**Issue by kkasravi** [kubeflow/kubeflow#2629]
Currently kfctl will open a file under bootstrap/config (depending on the init arguments) as part of it's initialization. These files are default.yaml, gcp_prototype.yaml, kfctl_basic_auth.yaml, kfctl_iap.yaml.
This information is read into the golang struct ComponentConfig 
```
type ComponentConfig struct {
	// Name of repository.
	Repo string `json:"repo,omitempty"`
	// List of default components.
	Components []string `json:"components,omitempty"`
	// List of default packages.
	Packages []string `json:"packages,omitempty"`
	// Parameters to be set to components using ks param set.
	ComponentParams Parameters `json:"componentParams,omitempty"`
	// Platform type.
	Platform string `json:"platform,omitempty"`
}
```

Currently this doesn't include any package manager information and should. 

The generated file 'app.yaml' that is produced after `kfctl init <name>` 
should also include package manager information so that subsequent 
subcommands `kfctl generate...`, `kfctl apply...` know which package manager to
delegate to.

The package manager stanza's need to include version information since each package manager
will download a repo using version information

ksonnet will download https://github.com/kubeflow/kubeflow/tarball/" + coordinator.Client.Spec.Version + "?archive=tar.gz

kustomize will download https://github.com/kubeflow/manifests/tarball/" + kustomize.Kustomize.Spec.Version + "?archive=tar.gz

**Comment by jlewi** [comment]
@kkasravi Any update on this?

Looks like #2685 may be related.

**Comment by jlewi** [comment]
@kkasravi Is this fixed by #2971?

**Comment by kkasravi** [comment]
yes
